portal/server/controllers/utilisation-report-service/utilisation-report-upload/index.js

```diff
@@ -131,7 +132,10 @@ const postUtilisationReportUpload = async (req, res) => {
       return renderPageWithError(req, res, extractDataErrorSummary, extractDataError, dueReportPeriods);
     }
 
-    const { csvValidationErrors } = await api.generateValidationErrorsForUtilisationReportData(csvJson, bankId, userToken);
+    // We filter down the JSON to only send the necessary data to the API for validation
+    const filteredCsvJson = filterReportJsonToRelevantKeys(csvJson);
```

Comment:
Can we ensure `filteredCsvJson` is not empty before invoking `generateValidationErrorsForUtilisationReportData` function.

Should we also check `bankId` and `userToken`?

Reply:
For bankId and userToken this comes from session where this isn't explicitly tested in a lot of places (and you can't really get to this page without having a a real session) so I think just having an error being caught later on and getting problem with service would be fine here.

I'll add an explicit check that csvJson isn't empty and throw an error if it is completely empty

Reply:
Thinking about this a bit more, all I'll be doing is adding throwing an error which will get caught and show the problem with service page, which is the same as the current behaviour. We don't have any requirements to show a special page to tell the user that the report can't be empty but I'm happy to bring this as a new requirement to UCD to think about?

Reply:
Sure, I'll throw an error and throw an error earlier on before we even filter the json to check it's not an empty array, an array of empty objects post filtering is actually still a valid payload

Reply:
Thank you, out of curiosity it is valid because it is an empty data set returned after removing frivolous dataset? If yes then should it proceed further or should we graceful throw an error back to the user?

Reply:
It's valid because we'll send it off to the API and it will return that there are missing required columns in the spreadsheet which is a valid user journey (and will show as missing column errors on the page back to the user)

Reply:
Understood, could you please add a line stating that a valid `[]` should be send to the below API to ensure validation errors are shown to the user at the front-end please.

Apart from this all looks good. Thanks